hw/drivers/flash/spiflash/src/spiflash.c

```diff
@@ -1143,6 +1155,72 @@ spiflash_sector_erase(struct spiflash_dev *dev, uint32_t addr)
                               &dev->characteristics->tse);
 }
 
+int
+spiflash_block_32k_erase(struct spiflash_dev *dev, uint32_t addr)
+{
+    return spiflash_erase_cmd(dev, SPIFLASH_BLOCK_ERASE_32KB, addr,
+                              &dev->characteristics->tbe1);
+}
+
+int
+spiflash_block_64k_erase(struct spiflash_dev *dev, uint32_t addr)
+{
+    return spiflash_erase_cmd(dev, SPIFLASH_BLOCK_ERASE_64KB, addr,
+                              &dev->characteristics->tbe2);
+}
+
+int
+spiflash_chip_erase(struct spiflash_dev *dev)
+{
+    uint8_t buf[1] = { SPIFLASH_CHIP_ERASE };
+
+    return spiflash_execute_erase(dev, buf, sizeof(buf),
+                                  &dev->characteristics->tce);
+}
+
+int
+spiflash_erase(struct spiflash_dev *dev, uint32_t address, uint32_t size)
+{
+    int rc = 0;
+
+    if (address == 0 && size == dev->hal.hf_size) {
+        return spiflash_chip_erase(dev);
+    }
+    address &= ~0xFFFU;
+    while (size) {
+        if ((address & 0xFFFFU) == 0 && (size >= 0x10000)) {
+            /* 64 KB erase if possible */
```

Comment:
are there flash which could not support block erase? If so maybe we should have MYNEWT_VAL which would enable block erase.